crates/gradbench/src/cli.rs

```diff
@@ -225,6 +230,18 @@ enum Message {
     },
 }
 
+impl Message {
+    /// Retrieve the `Id` from a message.
+    fn id(&self) -> Id {
+        match *self {
+            Message::Start { id } => id,
+            Message::Define { id, .. } => id,
+            Message::Analysis { of, .. } => of,
```

Comment:
This is wrong, yeah? An analysis message still has an `"id"` which is specifically different from the ID of the message it's an analysis `"of"`. Actually, I just realized that I forgot to write that in the protocol documentation, so I'll do that as well.